src/impl_/pyclass.rs

```diff
@@ -943,7 +943,15 @@ impl<T: PyClass> PyClassBaseType for T {
 
 /// Implementation of tp_dealloc for all pyclasses
 pub(crate) unsafe extern "C" fn tp_dealloc<T: PyClass>(obj: *mut ffi::PyObject) {
-    crate::callback_body!(py, T::Layout::tp_dealloc(obj, py))
+    /// Trampoline so that std::panic::catch_unwind only gets monomorphized once for
+    /// all tp_dealloc implementations
+    unsafe fn trampoline_dealloc(
```

Comment:
Why define that here instead of in `mod trampoline`?

Reply:
Because `T::Layout::tp_dealloc` takes `py` last it doesn't match the rest of the trampolines (which take `py` first). What I'll do is push a `dealloc` macro-generated trampoline, and leave a wrapper here along with a TODO here to clean up by rearranging arguments to `T::Layout::tp_dealloc`. (Which I think is follow-up-PR worthy as a breaking change.)